File: Bronze/2021-01/oddphotos_jan2021.java
import java.util.Scanner;

public class oddphotos_jan2021 {
    public static void main(String[] args) {
        // input
        Scanner sc = new Scanner(System.in);

        int numBreeds = sc.nextInt();
        int evenCnt = 0;
        int oddCnt = 0;
        for (int i=0; i<numBreeds; i++) {
            if (sc.nextInt()%2 == 0) {
                evenCnt++;
            } else {
                oddCnt++;
            }
        }

        // algorithm
        if (oddCnt > evenCnt) {
            // too many odd groups, merge two odd groups which form one even group
            while (oddCnt > evenCnt) {
                evenCnt++;
                oddCnt-=2;
            }
        }

        // output
        if (evenCnt > oddCnt+1) {
            System.out.println(2*oddCnt+1);
        } else {
            System.out.println(evenCnt+oddCnt);
        }
    }
}

File: Bronze/2021-01/oddphotos_jan2021.cpp
#include <iostream>

using namespace std;

int main() {
    int n;
    cin >> n;

    int cows[n];
    int odd = 0, even = 0;

    for (int i=0; i<n; i++) {
        cin >> cows[i];

        if (cows[i]%2 == 0) even++;
        else odd++;
    }

    while (odd > even) {
        // form two odd into one even
        even++;
        odd -= 2;
    }

    if (even > odd+1) cout << 2*odd+1 << endl; // if too many even, you can combine evens to get even
    else cout << odd + even << endl; // already processed if more odd, so just output
}